lib/gc/Transforms/Utils/ValueUtils.cpp

```diff
@@ -193,5 +198,126 @@ bool hasSharedMemSpace(mlir::Value memref) {
   return false;
 }
 
+std::tuple<SmallVector<Value>, Value>
+computeSubviewOffsets(PatternRewriter &rewriter, Location loc, Value memref) {
+  auto fillVal = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+  auto origShape = dyn_cast<MemRefType>(memref.getType()).getShape();
+
+  SmallVector<Value> resolvedOffsets(origShape.size(), fillVal);
+
+  while (auto subViewOp = memref.getDefiningOp<memref::SubViewOp>()) {
+    auto currentOffsets = getAsOpFoldResult(resolvedOffsets);
+    resolvedOffsets.clear();
+
+    affine::resolveIndicesIntoOpWithOffsetsAndStrides(
+        rewriter, memref.getLoc(), subViewOp.getMixedOffsets(),
+        subViewOp.getMixedStrides(), subViewOp.getDroppedDims(), currentOffsets,
+        resolvedOffsets);
+    memref = subViewOp.getOperand(0);
+  }
```

Comment:
the logic was taken from [`XeGPUFoldAliasOps` upstream pass](https://github.com/llvm/llvm-project/blob/2b3266c1701f315d7e89c81977800001563afacb/mlir/lib/Dialect/XeGPU/Transforms/XeGPUFoldAliasOps.cpp#L45-L56)